angular/src/app/app.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component } from '@angular/core';
+
+type Shape = {
+  x: number;
+  y: number;
+};
+
+@Component({
+  selector: 'app-root',
+  standalone: true,
+  template: `
+    <div id="wrapper">
+      @for (tile of this.tiles; track $index) {
+        <div
+          class="tile"
+          [style]="{ left: tile.x.toFixed(2) + 'px', top: tile.y.toFixed(2) + 'px' }"
+        ></div>
+      }
+    </div>
+  `,
```

Comment:
While this does not noticeably affect the loading times, the code creates one unnecessary `<div>` too many.

It should be written like this instead:
```suggestion
  host: {
    id: 'wrapper',
  },
  template: `
    @for (tile of this.tiles; track $index) {
      <div
        class="tile"
        [style]="{ left: tile.x.toFixed(2) + 'px', top: tile.y.toFixed(2) + 'px' }"
      ></div>
    }
  `,
```